components/org.wso2.carbon.identity.oauth.endpoint/src/main/java/org/wso2/carbon/identity/oauth/endpoint/authz/OAuth2AuthzEndpoint.java

```diff
@@ -1456,7 +1459,16 @@ private Response handleInitialAuthorizationRequest(OAuthMessage oAuthMessage) th
         }
         String redirectURL = handleOAuthAuthorizationRequest(oAuthMessage);
         String type = getRequestProtocolType(oAuthMessage);
-
+        try {
+            // Add the service provider id to the redirect URL. This is needed to support application wise branding.
+            String serviceProviderId =
+                    getServiceProvider(oAuthMessage.getRequest().getParameter("client_id")).getApplicationResourceId();
+            redirectURL = addServiceProviderIdToRedirectURI(redirectURL, serviceProviderId);
```

Comment:
`client_id`, lets get this to a constant, Check if there is an already a constant exists for this

Reply:
Addressed with https://github.com/wso2-extensions/identity-inbound-auth-oauth/pull/2588/commits/84fc98ab729c551156c0593622cdcd3f7cd52360